File: Lab2/lab3.py
#Task 1
pos=int(input("Enter positive integers greater than 1: "))
step=0
c=pos
while c>1:
    print(c, end=" -> ")
    if c%2==0:
        c=c//2
    else:
        c=3*c+1
    step+=1
print(1)
print("Total steps:", step)

#Task 2
while True:
    n=int(input("\nEnter possitive integers between 10 and 100"))
                
    if 10<=n2<=100:
        break
    print("Invalid output. Enter between 10 and 100")   

f_count=0
b_count=0
total_count=0
for i in range(1, n + 1):
    if i%7==0:
        continue
    if i%3==0 and i%5==0:
        print("FizzBuzz")
        total_count +=1
    elif i%3==0:
        print("Fizz")
        f_count += 1
    elif i%5==0:
        print("Buzz")
        b_count += 1
    else:
        print(i)
print("Summary")
print("Fizz count:", f_count)
print("Buzz count:", b_count)
print("FizzBuzz count:", total_count)

#Task 3
n = int(input("\nPlease enter a number between 3 and 9: "))
for i in range(1, 2 * n):
    k=n-abs(n-i) # 
    print("*"*k)

File: Lab2/lab3.cpp
#include <iostream>
#include <string>
#include <cmath>
using namespace std;
int main() {
    // TASK 1
    int pos, steps = 0;
    cout << "Enter positive integer greater than 1: ";
    cin >> pos;
    int current=pos;
    while (current>1){
        cout << current << " → ";
        if (current%2==0){
            current=current/2;
        } else{
            current=3*current+1;
        }
        steps++;
    }
    cout << 1 << endl;
    cout << "Total steps: " << steps << endl;
   // TASK 2
    int n;
    while (true) {
        cout << "\nPlease enter a number between 10 and 100: ";
        cin >> n;
        if (n >= 10 && n <= 100) {
            break;
        }
        cout << "Invalid input. ";
    }



    // TASK 3
    int n_star;
    cout << "Enter a number between 3 and 9: ";
    cin >> n_star;
    for (int i=1; i<=2*n_star-1; i++) {
        int k=n_star-abs(n_star - i);
        for (int j=0; j<k; j++) {
            cout << "*";
        }
        cout << endl;
    }
    return 0;
}
